include/fmt/ranges.h

```diff
@@ -414,20 +417,86 @@ struct range_formatter<
     if (it != end && *it == 'n') {
       set_brackets({}, {});
       ++it;
+    } else {
+      bool check_for_s = false;
+      if (it != end && *it == '?') {
+        ++it;
+        detail::maybe_set_debug_format(underlying_, true);
+        set_brackets({}, {});
+        check_for_s = true;
+        is_debug = true;
+      }
+      if (it != end && *it == 's') {
+        if (!std::is_same<T, Char>::value) {
+          report_error("invalid format specifier");
+        }
+        if (!is_debug) {
+          set_brackets(detail::string_literal<Char, '"'>{},
+                       detail::string_literal<Char, '"'>{});
+        }
+        check_for_s = false;
+        is_string_format = true;
+        ++it;
+      }
+      if (check_for_s) {
+        report_error("invalid format specifier");
+      }
     }
 
     if (it != end && *it != '}') {
-      if (*it != ':') report_error("invalid format specifier");
+      if (is_string_format || *it != ':')
```

Comment:
`is_string_format` check is not needed, just return after `s` has been parsed above.

Reply:
Same idea for the comment above about disallowing `n` with `s` and `?s`,  this was meant to disallow `range-underlying-spec` given `s` or `?s`.